Blueprint crud

rizenfrmtheashes opened the pull request:
Update column is almost done, going to push when that's done, then going to push testing to confirm working state.

Commit:
Add xForwardedFor to event body

Commit:
Start work on validator with testing

Commit:
Start work on validator with testing

Commit:
An Early Branch

Commit:
An Early Branch

Commit:
Update function headers and set up overall structure

Commit:
Update function headers and set up overall structure

Commit:
Add First working(?) test

Commit:
Add First working(?) test

Commit:
Update test and fix bugs to get simple add table working

Commit:
Update test and fix bugs to get simple add table working

Commit:
Encapsulate column collision check and self modification into method

Commit:
Encapsulate column collision check and self modification into method

Commit:
Add 'Remove Column' in 'Update Table' operation

Commit:
Add 'Remove Column' in 'Update Table' operation

Commit:
Start work on 'update column' operation

Commit:
Start work on 'update column' operation

Commit:
Add '.arcconfig' to .gitignore

Commit:
Add '.arcconfig' to .gitignore

GarethLewin commented on dynamo_connector/dynamo_connector.go:
```diff
@@ -0,0 +1,12 @@
+package dynamo_connector
```
might want to name this package dynamoConnector or similar.

aasted replied:
(don't forget to change the package name)

Commit:
Update validation to include finished 'Update Column'

Commit:
Update validation to include finished 'Update Column'

Commit:
Refactor column operations to prevent split validation process

Commit:
Refactor column operations to prevent split validation process

aasted committed:
Merge pull request #8 from TwitchScience/add_xforwarded

Add xForwardedFor to event body

aasted commented on dynamo_connector/dynamo_connector.go:
```diff
@@ -0,0 +1,12 @@
+	"github.com/twitchscience/scoop_protocol/schema_storer"
+)
+
+type DynamoConnector interface {
```
Why is schema_storer.TableSchema not named but tableName is?

aasted commented on schema/schema.go:
```diff
@@ -1,15 +1,167 @@
+	return false
+}
+
+func (e *Event) AddColumn(ColumnOperation ColumnOperation) error {
```
The uppercase name here shadows the type; I think you want the value to be columnOperation. You repeat this pattern elsewhere.

aasted commented on schema/schema.go:
```diff
@@ -1,15 +1,167 @@
+
+func (e *Event) AddColumn(ColumnOperation ColumnOperation) error {
+
+	// column operation is add
```
this comment is superfluous; the code is obvious, and your error message says what we're checking.

aasted commented on schema/schema.go:
```diff
@@ -1,15 +1,167 @@
+func (e *Event) AddColumn(ColumnOperation ColumnOperation) error {
+
+	// column operation is add
+	if ColumnOperation.Operation != "add" {
```
Idle thought: This pattern makes me wonder if columnOperation is the right type, or if it should have some wrapped inner value that doesn't have the Operation field. This consistency checking makes it feel like the type is too big -- the caller shouldn't have to think about setting this value, it's an implementation detail of our handlers.

aasted commented on schema/schema_migrator.go:
```diff
@@ -0,0 +1,108 @@
+	}
+	//checks for existance of atleast single distKey
+	if len(m.possibleMigration.TableOption.DistKey) < 1 {
+		return nil, errors.New("Tableoption must contain at least 1 distkey")
```
TableOption for consistency

aasted commented on schema/schema_test.go:
```diff
@@ -0,0 +1,295 @@
+	if err != nil {
+		t.Log(err.Error())
+		t.Log("err was incorrect, should have passed")
+		t.Fail()
```
Above 3 lines can be simplified to:
t.Fatalf("%s\nerr was incorrect, should have passed", err.Error())

Might read better as: "Unexpected error: %s"

aasted commented on schema/schema_test.go:
```diff
@@ -0,0 +1,295 @@
+	}
+
+	//Test if table already exists.
+	t.Log("Testing 'Table already exists' check")
```
This looks like a new test.

aasted commented on schema/schema_test.go:
```diff
@@ -0,0 +1,295 @@
+	}
+}
+
+func TestAddTable(t *testing.T) {
```
Consider factoring out your boilerplate and running these as separate tests so that if multiple fail they don't prevent each other from telling you what's wrong.

Commit:
Update migrator tests to be modular

Commit:
Update migrator tests to be modular

Commit:
Create test objects in test package for multi-package testing

Commit:
Create test objects in test package for multi-package testing

Commit:
Update schema objects for handler test

Commit:
Update schema objects for handler test

Commit:
Add test objects and fix remove bug

Commit:
Add test objects and fix remove bug

Commit:
Add comments to better explain migrator checks

Commit:
Start work on validator with testing

Summary:
An Early Branch

Update function headers and set up overall structure

Add First working(?) test

Update test and fix bugs to get simple add table working

Encapsulate column collision check and self modification into method

Add 'Remove Column' in 'Update Table' operation

Start work on 'update column' operation

Add '.arcconfig' to .gitignore

Update validation to include finished 'Update Column'

Refactor column operations to prevent split validation process

Update migrator tests to be modular

Create test objects in test package for multi-package testing

Update schema objects for handler test

Add test objects and fix remove bug

Add comments to better explain migrator checks

Test Plan: Crap ton of unit tests, integration incoming

Reviewers: gareth

Subscribers: mperrone, maasted

Differential Revision: https://phabricator.internal.justin.tv/D2785

Commit:
Add centralized errprs

Commit:
Update for bugfixes and file sizes

Commit:
Update errors and test to be more correct

Commit:
Update exported functions with comments so golint passes

Commit:
Include comments in migrator file

Commit:
Update conversion to pass gometalinter

Commit:
Update migrator to return event val instead of pointer

Commit:
Fix merge conflicts with remote branch

Commit:
Add TableVersion to RowCopyRequest

Commit:
Update to fix changing wrong struct
